src/persist-client/src/operators/shard_source.rs

```diff
@@ -278,7 +278,7 @@ where
     }
     impl<H: FnOnce(String) -> Pin<Box<dyn Future<Output = ()>>> + 'static> ErrorHandler<H> {
         /// Report the error and enforce that we never return.
-        async fn report_and_stop(self, error: String) {
+        async fn report_and_stop(self, error: String) -> ! {
```

Comment:
ha! does this actually work now? I suggested this to @petrosagg but he thought diverging functions would only be available in nightly.

Reply:
The never type [can only be used as a return type](https://doc.rust-lang.org/reference/types/never.html), but it could be used as that since ~forever!